src/components/Image/Image.jsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from './Image.scss';
+
+function ImageAmp ( {
+	alt,
+	height,
+	sizes,
+	src,
+	srcSet,
+	title,
+	width,
+} ) {
+	return (
+		<amp-img
+			alt={alt}
+			height="1"
+			layout="responsive"
+			sizes={sizes}
+			src={src}
+			srcSet={srcSet}
+			title={title}
+			width={width / height}
+		/>
+	);
+};
+
+ImageAmp.propTypes = {
+	alt: PropTypes.string.isRequired,
+	height: PropTypes.number.isRequired,
+	sizes: PropTypes.string.isRequired,
+	src: PropTypes.string.isRequired,
+	srcSet: PropTypes.string.isRequired,
+	title: PropTypes.string,
+	width: PropTypes.number.isRequired,
+};
+
+function Image ( {
+	alt,
+	amp,
+	height,
+	loading,
+	sizes,
+	src,
+	srcSet,
+	title,
+	width,
+} ) {
+	if ( amp ) {
+		return (
+			<ImageAmp
+				alt={alt}
+				height={1}
+				sizes={sizes}
+				src={src}
+				srcSet={srcSet}
+				title={title}
+				width={width / height}
+			/>
+		);
+	}
+
+	return (
+		<img
+			alt={alt}
+			className={styles.image}
+			decoding="async"
+			height={height}
+			loading={loading}
+			sizes={sizes}
+			src={src}
+			srcSet={srcSet}
+			title={title}
+			width={width}
+		/>
+	);
+};
+
+Image.propTypes = {
+	/**
+	 * Alternative text to describe the image for screen readers or in situations
+	 * where the image cannot be loaded. This prop is required but under certain
+	 * circumstances an empty string is preferred.
+	 */
+	alt: PropTypes.string.isRequired,
+
+	/**
+	 * Whether to render the AMP version of the image.
+	 */
+	amp: PropTypes.bool.isRequired,
+
+	/**
+	 * The height of the image. The absolute value is not as important as its
+	 * correspondences with the `width` prop to set the correct aspect ratio. A
+	 * good practice is to set this at the largest size at which this image can be
+	 * rendered.
+	 */
+	height: PropTypes.number.isRequired,
+
+	/**
+	 * How and when the image should be loaded—e.g., right away or only after the
+	 * image is revealed in the viewport.
```

Comment:
Suggest linking to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img to make it clear that the available values aren't decided by us